src/internal/SignalGenerator.cpp

```diff
@@ -0,0 +1,52 @@
+/**
+* @file SignalGenerator.cpp
+* @brief This header file contains the implementation functions of the SignalGenerators class.
+*
+* @author Murilo Mendonça Venâncio
+* 
+* @date 24/01/2019
+*/
+
+#include "SignalGenerator.hpp"
+
+
+SignalGenerator::SignalGenerator()
+{
```

Comment:
As @joaobrittoneto mentioned before, we usually initialize the parameters as show below:

SignalGenerator::SignalGenerator()
  : m_amp(1)
  , m_freq(1)
  , m_cosine(true)
{
}